freqtrade/persistence/models.py

```diff
@@ -259,9 +263,79 @@ class LocalTrade():
     strategy: str = ''
     timeframe: Optional[int] = None
 
+    # Margin trading properties
+    borrowed_currency: str = None
+    collateral_currency: str = None
+    interest_rate: float = 0.0
+    liquidation_price: float = None
+    is_short: bool = False
+    borrowed: float = 0.0
+    leverage: float = None
+
+    # @property
+    # def base_currency(self) -> str:
+    #     if not self.pair:
+    #         raise OperationalException('LocalTrade.pair must be assigned')
+    #     return self.pair.split("/")[1]
+
+    # TODO: @samgermain: Amount should be persisted "as is".
```

Comment:
i don't think you understood the intend with "amount".
it defines what you own on the exchange. Leverage does not really play a role in this - and as such, this part is definitely wrong.

I can buy 100 XRP  (then i own 100 XRP) - i can buy 200 XRP with 2x leverage - then i own 200 XRP - and that's what goes into amount.

"leverage" only comes into play once you define how much you paid for this.

Reply:
I was under the impression that when you send an order to Kraken, you send the amount without leverage applied, and also send the leverage. This might still be the case, but Im not sure because I made [an order with leverage](https://i.imgur.com/kCsThGe.png), and [one without leverage](https://i.imgur.com/ZEOV6TE.png) and they're giving the same values. So far my balance has only updated for one of the sales worth 0.0001, but I think they've both been completed. Kraken seems to give me errors and not explain them.

Reply:
I think that the way it works on kraken is that you specify the amount like 0.0001 btc with 3x leverage, and then the profits you make are as if you had bought 0.0003 btc.

With this logic, any time you access `amount` it gives you `_amount * leverage` I tried to abstract away the use of all the `_` properties as much as I could, so that the only things that reference them are the computed properties

 If you're trading on an exchange that doesn't work like this, then you would pass `borrowed` and `amount` to trade and when you access `amount` it just gives you `_amount`
 
 If you can think of a better way to do this I could do that, but I was trying to abstract away the implementation details of how different exchanges handle the amount you trade vs your leverage. This is also why I made only one of `leverage` and `borrowed` able to be set.
 
 We could maybe create a separate class that contains these 3 variables to abstract the logic away further?

Reply:
I actually might be wrong about how kraken determines your leverage/borrowed amount(Kraken might do it in a way where they divide your stake amount by your leverage, trying to figure it out) but it looks like it's kinda different for each exchange. I would like to keep the computed properties if we could because doing something like 

```
@property
def leverage(self):
    if (_leverage):
        return _leverage
    else:
        return exchange.getLeverage(_borrowed, _amount)
```

or even just

```
@property
def leverage(self):
        return exchange.getLeverage(_leverage, _borrowed, _amount)
```

I thought about just assigning these properties in the constructor, but then if we have anything where we are changing the `amount` `leverage` or `borrowed`, then we won't be able to calculate the new numbers because the original numbers won't make sense anymore.  I also decided to do it this way when I was still passing `leverage` and `borrowed` through the `update` function

It's set up like this so that the `_` properties are always the original values the properties without `_` are as we understand them